crates/services/p2p/src/service.rs

```diff
@@ -805,6 +810,7 @@ where
             PostcardCodec::new(max_block_size),
         )
         .await?;
+        p2p_service.update_block_height(last_height);
```

Comment:
On my laptop yes, here we have 4 CI without error something that hasn't happened since long time. Even if there is one that will still fails after we could reopen an issue to try to fix it again but I think it's already a big improvement that we can bring to master